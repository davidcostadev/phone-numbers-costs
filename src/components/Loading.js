import React from 'react';
import classname from 'classnames';
import PropTypes from 'prop-types';

const Loading = ({ children, loading }) => (
  <div className="page">
    <div className={classname(
      'page__loader',
      { 'page__loader--loading': loading },
    )}
    >
      loading...
    </div>
    {children}
  </div>
);

Loading.propTypes = {
  children: PropTypes.any.isRequired,
  loading: PropTypes.bool.isRequired,
};

export default Loading;
