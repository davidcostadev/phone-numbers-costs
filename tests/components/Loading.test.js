import React from 'react';
import renderer from 'react-test-renderer';
import Loading from '../../src/components/Loading';

describe('Loading Component', () => {
  describe('Render', () => {
    it('is loading', () => {
      const loading = true;
      const tree = renderer.create((
        <Loading loading={loading}>
          <div>children</div>
        </Loading>
      )).toJSON();

      expect(tree).toMatchSnapshot();
    });

    it('not loading', () => {
      const loading = false;
      const tree = renderer.create((
        <Loading loading={loading}>
          <div>children</div>
        </Loading>
      )).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
