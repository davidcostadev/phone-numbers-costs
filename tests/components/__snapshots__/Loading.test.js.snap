// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Component Render is loading 1`] = `
<div
  className="page"
>
  <div
    className="page__loader page__loader--loading"
  >
    loading...
  </div>
  <div>
    children
  </div>
</div>
`;

exports[`Loading Component Render not loading 1`] = `
<div
  className="page"
>
  <div
    className="page__loader"
  >
    loading...
  </div>
  <div>
    children
  </div>
</div>
`;
